### PR TITLE
Add Reader Writer Reentrant lock to avoid CMException  in BufferManagerImpl

### LINKED ISSUE #1418: Possible CMException with BufferManagerImpl when forwarding to Sync Handler
Possible CMException with BufferManagerImpl (Conduit) when forwarding to log events to a synchronous handler. The BufferManagerImpl.add() method is called by Source classes (I.e LogSource, TraceSource, etc)  and will go through a set of synchronized handlers to forward too. If the set is modified then a CMException can occur.

This was identified in the code review of Pull Request 1285

### PR DESCRIPTION
**Pull request by Channyboy**
Add Reader Writer Re-entrant lock to avoid CMException when accessing/modifying synchronized handler set.

Fixes #1418

### CODE REVIEW

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
The failing FAT has no relation to the changes made in this PR

File: dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java [hunk 1]
```diff
@@ -107,11 +117,31 @@ public void addHandler(String handlerId) {
...
+        /* There can be many Reader locks, but only one writer lock.
+        *  This ReaderWriter lock is needed to avoid CMException when the add() method is forwarding log events
+        *  to synchronized handlers and an addSyncHandler or removeSyncHandler is called
+        */
+		RERWLOCK.writeLock().lock();
+		try{
+			synchronizedHandlerSet.add(syncHandler);
+		} finally {
+        RERWLOCK.writeLock().unlock();
+    	}
```
Comment (donbourne) [inline dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java]: indentation problems above...

File: dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java [hunk 2]
```diff
@@ -107,11 +117,31 @@ public void addHandler(String handlerId) {
...
 	}
 	
 	public void removeSyncHandler(SynchronousHandler syncHandler) {
-		synchronizedHandlerSet.remove(syncHandler);
+        /*
```
Comment (donbourne) [inline dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java]: indentation

**Pull request merged**